platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LogoutServiceImpl.java

```diff
@@ -65,23 +65,32 @@ public String getActionProviders(HttpServletRequest request, HttpServletResponse
     subjectMap.put("http_response", response);
     subjectMap.put(SecurityConstants.SECURITY_SUBJECT, subject);
 
-    List<Map<String, String>> actionPropertiesList = new ArrayList<>();
+    Map<String, String> actionProperties = new HashMap<>();
 
     for (ActionProvider actionProvider : logoutActionProviders) {
-      Map<String, String> actionProperties = new HashMap<>();
       Action action = actionProvider.getAction(subjectMap);
 
       if (action != null) {
         String displayName = SubjectUtils.getName(subject, "", true);
         actionProperties.put("title", action.getTitle());
         actionProperties.put("auth", displayName);
         actionProperties.put("description", action.getDescription());
-        actionProperties.put("url", action.getUrl().toString());
-        actionPropertiesList.add(actionProperties);
+
+        StringBuilder urlBuilder = new StringBuilder(action.getUrl().toString());
+        String referer = request.getHeader("Referer");
+        if (!StringUtils.isEmpty(referer)) {
+          String[] parts = referer.split("\\?prevurl=");
+          if (parts.length > 1) {
+            String previousUrl = parts[1];
+            urlBuilder.append("?prevurl=");
```

Comment:
❓ What about the `service` query parameter? The Admin UI and Intrigue use it instead of `prevurl`

Reply:
It's all switched to prevurl in the js code.